src/main/java/Duke.java

```diff
@@ -1,10 +1,193 @@
+            System.out.println(INVALID_COMMAND_PRINTER);
+        }else{
+            tasks.get(index).setDone(true);
+            System.out.println(LINE);
+            System.out.println(VALID_COMMAND_PRINTER);
+            System.out.println("Task has been marked as: completed");
+            System.out.println(tasks.get(index).toString());
+            System.out.println(LINE);
+        }
+    }
+    private static void unmark(String[] commands){
+        if (commands.length!=2){
+            System.out.println(LINE);
+            System.out.println("Please only input \"unmark\" followed by an integer.");
+            System.out.println(INVALID_COMMAND_PRINTER);
+            System.out.println(LINE);
+            return;
+        }
+        int index;
+        try{
+            index = Integer.parseInt(commands[1]);
+        } catch (NumberFormatException numberFormatException){
+            System.out.println(LINE+'\n'+"Sorry, index is not numeric."+'\n'+LINE);
+            System.out.println(INVALID_COMMAND_PRINTER);
+            return;
+        }
+        index--;
+        if(index<0||index>=tasks.size()){
+            System.out.println(LINE+'\n'+"Sorry, index is invalid."+'\n'+LINE);
+            System.out.println(INVALID_COMMAND_PRINTER);
+        }else if(!tasks.get(index).isDone()){
+            System.out.println(LINE+'\n'+"The task specified is already unmarked."+'\n'+LINE);
+            System.out.println(INVALID_COMMAND_PRINTER);
+        }else{
+            tasks.get(index).setDone(false);
+            System.out.println(LINE);
+            System.out.println(VALID_COMMAND_PRINTER);
+            System.out.println("Task has been marked as: not completed");
+            System.out.println(tasks.get(index).toString());
+            System.out.println(LINE);
+        }
+    }
+    private static taskType getTaskType(String command){
+        String[] commands = command.split(" ");
+        if(commands[0].equals("deadline")){
+            return taskType.DEADLINE;
+        }else if(commands[0].equals("event")){
+            return taskType.EVENT;
+        }else if(commands[0].equals("todo")){
+            return taskType.TODO;
+        }
+        return taskType.INVALID;
+    }
+    private static boolean checkCommandValidity(String command, taskType currentTaskType){
+        if(currentTaskType==taskType.DEADLINE){
+            if(command.contains("/by")){
+                return true;
+            }
+            return false;
+        }else if(currentTaskType==taskType.EVENT){
+            if(command.contains("/from")&&command.contains("/to")){
+                return true;
+            }
+            return false;
+        }else if(currentTaskType==taskType.TODO){
+            if(command.split(" ").length<=1){
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
+    private static void addTask(String command){
+        taskType currentTaskType = getTaskType(command);
```

Comment:
I like the correct usage of camelCase in variables and method names while PascalCase is used in Class names.